lib/worker.js

```diff
@@ -84,30 +80,78 @@ async function getLocalLinter(basedir) {
             linter = require('loophole').allowUnsafeNewFunction(() => require(linterPath).Linter);
           }
         } else {
-          linter = tslintDef;
+          return resolve();
         }
-        tslintCache.set(basedir, linter);
+        tslintCache.set(fileDir, linter);
```

Comment:
❓ Looks like @Xapphire13's earlier comment got marked resolved without being resolved:
Shouldn't this be `baseDir`?

Reply:
Actually, that's correct as-is (at least, it's what I intended, "correct" is rather subjective 😛.) It specifically needs to be fileDir because the lookup on lines 110-113 searches the cache for the linter that was used the last time a source file *in that directory* was linted. If we use baseDir, which might actually be the user's `npm` directory, then it'll look up which linter to use every time and it will never actually get cached in a way that it can find later.